workers/unity/Packages/com.improbable.gdk.buildsystem/WorkerBuilder.cs

```diff
@@ -165,12 +194,14 @@ private static bool BuildWorkers(string[] workerTypes, BuildEnvironment buildEnv
             }
         }
 
-        private static bool BuildWorkerForEnvironment(string workerType, BuildEnvironment targetEnvironment, ScriptingImplementation? scriptingBackend = null)
+        private static bool BuildWorkerForEnvironment(string workerType, BuildEnvironment targetEnvironment, BuildTarget buildTarget, ScriptingImplementation? scriptingBackend = null)
         {
             var spatialOSBuildConfiguration = BuildConfig.GetInstance();
             var environmentConfig = spatialOSBuildConfiguration.GetEnvironmentConfigForWorker(workerType, targetEnvironment);
 
-            var enabledTargets = environmentConfig?.BuildTargets.Where(t => t.Enabled).ToList();
+            var enabledTargets = buildTarget == BuildTarget.NoTarget
```

Comment:
This feels like a good time to refactor into:

```csharp
bool BuildWorkerForEnvironment(string workerType, BuildEnvironment targetEnvironment, ScriptingImplementation? scriptingBackend = null);

bool BuildWorkerForEnvironmentAndTarget(string workerType, BuildEnvironment targetEnvironment, BuildTarget target, ScriptingImplementation? scriptingBackend = null);
```

then call the correct one according to if we are looking for a particular target over a all targets 

(If you end up adding comma-delineated support, you'll want to make it a `IEnumerable<BuildTarget>` but a similar argument applies)